proposals/csharp-11.0/low-level-struct-improvements.md

```diff
@@ -1810,47 +1810,27 @@ void Usage()
 
 To make these APIs usable the compiler ensures that the `ref` lifetime for a `ref` parameter is smaller than lifetime of any references in the associated parameter value. This is the rationale for having *ref-safe-context* for `ref` to `ref struct` be *return-only* and `out` be *caller-context*. That prevents cyclic assignment because of the difference in lifetimes.
 
-It is also why `[UnscopedRef]` only promotes the *ref-safe-context* of any `ref` to `ref struct` values to *return-only* and not *caller-context*. Consider that using *caller-context* allows for cyclic assignment and would force a viral use of `[UnscopedRef]` for a `ref struct`:
+Note that `[UnscopedRef]` promotes the *ref-safe-context* of any `ref` to `ref struct` values to *caller-context*:
 
 ```c#
-ref struct S
-{
-    byte Field;
-
-    [UnscopedRef]
-    public Span<byte> Data => new Span<byte>(ref Field, 1);
-}
-
-void M(ref S s)
+S F()
 {
-    // Error: passing a scoped ref to [UnscopedRef] ref 
-    Span<byte> span = s.Data;
+    S local = new();
+    // Error: self assignment possible inside `S.M`.
+    S.M(ref local);
+    return local;
 }
-```
-
-This is correctly illegal in that case because the compiler has to consider the pathological case that `S.Data` could cyclic assign via `this`. That forces methods all methods that call `S.Data` to further mark their `ref` parameters as `[UnscopedRef]`. This is viral until the method which creates the value as a local. This is why *return-only* exists as a *safe-context*. It does complicate the spec / implementation but it serves to make the feature significantly more usable.
 
-Note: this cyclic assignment problem does continue to exist for `[UnscopedRef] out` to `ref struct` because that causes the *safe-context* and *ref-safe-context* to be equivalent. 
-
-```c#
-ref struct RS
+ref struct S
 {
     int field;
     ref int refField;
-}
-
-void M1(out RS p)
-{
-    // Error: from method arguments must match:
-    // Step 1 would calculate the narrowest escape as *caller-context*
-    // Step 2 would fail the assignment check because p has safe-context of *return-only*
-    M2(out p);
-}
 
-void M2([UnscopedRef] out RS p)
-{
-    // The lifetimes of LHS and RHS are equivalent here and hence this is legal
-    p.refField = ref p.Field;
+    static void M([UnscopedRef] ref S s)
+    {
+        // Allowed: s has both safe-context and ref-safe-context of caller-context
+        s.refField = ref s.field;
+    }
 }
```

Comment:
Think that we should have a sample where `[UnscopedRef]` to assign a `ref` field. Basically demonstrate that this behavior we chose provides value when the type is not a `ref struct`. Essentially: 

```csharp
ref struct RS
{
   ref int refField;

   public void Assign([UnscopedRef] int data)
   {
      refField = ref data;
   }
}
```